src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.js

```diff
@@ -50,6 +50,14 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory {
           ...parsedUiState,
           ...panel.embeddableConfig,
         });
+        container.onEmbeddableConfigChanged(newEmbeddableConfig => {
```

Comment:
This method currently only sets all values from the new state. Don't we also need to clean up the values in the `uiState`, that are not present in `newEmbeddableConfig` anymore, so they won't accidentally stay in the `uiState`?

It seems we currently don't have a `clear` method on the `uiState` which would be tremendous useful in that case. Maybe we could introduce it with this PR too?